src/system.jl

```diff
@@ -25,6 +25,47 @@ end
 const Zones = Dictionary{ZoneNum, Zone}
 
 const UnitCode = Int64
+const GenId = Union{UnitCode, InlineString31}
+
+"""
+    $TYPEDEF
+
+Abstract type to describe generation-specific internals of generators.
+"""
+abstract type Technology end
+
+"""
+    $TYPEDEF
+
+Type to describe a single configuration of a combined cycle generator.
+
+Fields:
+$TYPEDFIELDS
+"""
+Base.@kwdef struct CombinedCycle <: Technology
+    "Id linking back to the resource this configuration is from"
+    parent::GenId
+    "Symbol describing the technology of the generator"
+    fuel_type::Symbol
+end
+# Handle string conversion
+CombinedCycle(parent::AbstractString, args...) = CombinedCycle(InlineString31(parent), args...)
```

Comment:
Ugly code to avoid a breaking change. I'm inclined just to remove this and only allow string ids instead